src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,8 +289,49 @@ def algorithm(self, parameters=None):
             with CodeTimer("Rucio block resolution", logger=logging):
                 self.findRucioBlocks(uniqueActivePU, msPileupList)
 
-            # TODO in future tickets: find the json files in the spec/sandbox area and tweak them
+            # define where to look for sandbox tar balls
+            idir = getattr(self.config.WorkflowUpdater, "componentDir")
```

Comment:
Please move this to the `__init__` method.

Reply:
why? The point is that I don't know what happen at run-time and when this area is created. In other words, does it part of WMA installation or is it part of run-time, i.e. when first job is processed. I think it belongs in a right place since I line below I check if it exists, and use `'/data/srv/wmagent/current/install/wmagentpy3'` as fall back area. And if neither exist the algorithm cycle fails with proper log error.

Reply:
if this attribute does not exist, the component will actually fail to even start. This is not a runtime attribute, it is a deployment configuration/attribute. In other words, it is defined only once in an agent lifetime.

Reply:
Alan, it is not about component but rather about directory. Does directory exist before run time?

Reply:
Yes, it does. I am only saying that whenever you read the component configuration, the ideal place for that is in the __init__ method (or setup, when it exists).